packages/apollo-gateway/src/index.ts

```diff
@@ -82,6 +126,16 @@ export class ApolloGateway implements GraphQLService {
   private engineConfig: GraphQLServiceEngineConfig | undefined;
   private pollingTimer?: NodeJS.Timer;
   private onSchemaChangeListeners = new Set<SchemaChangeCallback>();
+  private serviceDefinitions: ServiceDefinition[] = [];
+
+  // Observe query plan, service info, and operation info prior to execution. The information made available here will give insight into the resulting query plan and the inputs that generated it.
```

Comment:
nit: I think usually comment strings still abide by column limits? (i.e. we should break this into multiple lines)

Reply:
I wonder if we want to make these async callbacks and then specifically not await them. Asking since the typical user experience will probably be to send metrics to their own systems about this information, which they'll likely have to make non-awaited async calls to themselves.

Reply:
We can also make it either return a `Promise<void> | void`, and if it's the latter it will be synchronous. I'm not sure what the best practice is here, to be honest, and am pretty open to suggestions, but just wanted to throw out some ideas.

Reply:
We're only awaiting the service definition updater right now (because we have to). I'm not sure what else you're referring to